src/clasqa/QADB.groovy

```diff
@@ -475,6 +492,47 @@ class QADB {
   public void resetAccumulatedCharge() { chargeTotal = 0 }
 
 
+  //..................................................
+  // Faraday Cup Helicity-Latched charge accumulator
+  //``````````````````````````````````````````````````
+  // -- accumulator
+  // call this method after evaluating QA cuts (or at least after calling query())
+  // to add the current file's Helicity-Latched charge to the total;
+  public void accumulateChargeHL() {
+    if(!chargeHLCounted) {
+      if(!( [runnum,filenum] in chargeHLCountedFiles )) {
+        chargeHLTotalMinus += chargeHL["-1"]
+        chargeHLTotalZero += chargeHL["0"]
+        chargeHLTotalPlus += chargeHL["1"]
+        chargeHLCountedFiles << [runnum,filenum]
+      }
+      chargeHLCounted = true
+    }
+  }
+  // -- accessor
+  // call this method at the end of your event loop
+  // - input is the helicity state (-1,0,1)
+  public double getAccumulatedChargeHL(int state) {
+    def ret = -1
+    if(state==-1){
+      ret = chargeHLTotalMinus
+    }else if(state==0){
+      ret = chargeHLTotalZero
+    }else if(state==1){
+      ret = chargeHLTotalPlus
+    }else{
+      System.err << "Invalid helicity state $state.  Use -1, 0, or 1 \n"  
```

Comment:
Might as well throw an exception if the input is wrong; then you don't have to return `-1`. Something like:
```suggestion
    throw new Exception("Invalid helicity state $state.  Use -1, 0, or 1")  
```

Reply:
Sorry, I meant `RuntimeException`, since that'll exit nonzero. I'll push a commit to take care of that.